crates/pyhq/python/hyperqueue/ffi/protocol.py

```diff
@@ -1,12 +1,24 @@
 import dataclasses
-from typing import Dict, List, Optional, Sequence
+from typing import Dict, List, Optional, Sequence, Union
 
 
 @dataclasses.dataclass()
 class ResourceRequest:
     n_nodes: int = 0
-    cpus: str = "1"
-    generic: Dict[str, int] = dataclasses.field(default_factory=dict)
+    resources: Dict[str, Union[int, str]] = dataclasses.field(default_factory=dict)
+
+    def __init__(
```

Comment:
Dataclasses shouldn't have an `__init__` method. I would turn this into a `@staticmethod` constructor. We can add an assert in `__post_init__` there there is a CPU resource.

Reply:
I removed data class from resource request